app/src/main/java/com/github/se/studybuddies/ui/chat/ContactScreen.kt

```diff
@@ -0,0 +1,247 @@
+                    },
+                    onCancel = { isDeleteContactDialogVisible = false })
+              }
+            }
+      }
+}
+
+@Composable
+private fun ToggleMapVisibilityButton(showOnMapState: MutableState<Boolean>) {
+
+  if (showOnMapState.value) {
+    Button(
+        onClick = { showOnMapState.value = false },
+        modifier = Modifier.width(300.dp).height(50.dp).clip(RoundedCornerShape(4.dp)),
+        colors = ButtonDefaults.buttonColors(containerColor = Blue, contentColor = White)) {
+          Text(text = stringResource(R.string.location_shared))
+        }
+  } else {
+    Button(
+        onClick = { showOnMapState.value = true },
+        modifier = Modifier.width(300.dp).height(50.dp).clip(RoundedCornerShape(4.dp)),
+        colors =
+            ButtonDefaults.buttonColors(containerColor = Color.LightGray, contentColor = White)) {
+          Text(text = stringResource(R.string.location_not_shared))
+        }
+  }
+}
+
+@Composable
+private fun DeleteContactDialog(onDelete: () -> Unit, onCancel: () -> Unit, contactID: String) {
+  Dialog(onDismissRequest = {}) {
+    Box(
+        modifier =
+            Modifier.width(300.dp)
+                .height(270.dp)
+                .clip(RoundedCornerShape(12.dp))
+                .background(Color.White)
+                .testTag(contactID + "_delete_box")) {
+          Column(
+              modifier = Modifier.padding(16.dp).testTag(contactID + "_delete_column"),
+              verticalArrangement = Arrangement.Center,
+              horizontalAlignment = Alignment.CenterHorizontally) {
+                Text(
+                    text = stringResource(R.string.warning_1_contact_deletion),
+                    modifier = Modifier.testTag(contactID + "_delete_text1"),
+                    color = Blue,
+                    textAlign = TextAlign.Center)
+                Text(
+                    text = stringResource(R.string.warning_2_contact_deletion),
+                    modifier = Modifier.testTag(contactID + "_delete_text2"),
+                    color = Blue,
+                    textAlign = TextAlign.Center)
+                Spacer(modifier = Modifier.height(20.dp))
+
+                DeleteButton(onClick = { onDelete() })
+                Spacer(modifier = Modifier.height(20.dp))
+
+                Button(
+                    onClick = { onCancel() },
+                    modifier =
+                        Modifier.clip(RoundedCornerShape(4.dp))
+                            .width(100.dp)
+                            .height(40.dp)
+                            .testTag(contactID + "_delete_no_button"),
+                    colors =
+                        ButtonDefaults.buttonColors(containerColor = Blue, contentColor = White)) {
+                      Text(
+                          text = stringResource(R.string.cancel),
+                          modifier = Modifier.testTag(contactID + "_delete_no_text"))

```

Comment:
use string

Reply:
might have time to reformat this next PR